pyproject.toml

```diff
@@ -67,68 +70,65 @@ classifiers = [
     "Typing :: Typed",
 ]
 
-requires-python = ">=3.6.2"
+include = ["resources/"]
 
-dependencies = [
-    "boto3",
-    "click",
-    "cloup",
-    "deepdiff",
-    "dnspython",
-    "file_read_backwards",
-    "humanfriendly",
-    "jinja2",
-    "kazoo",
-    "lxml",
-    "psutil",
-    "pygments",
-    "pyopenssl",
-    "python-dateutil",
-    "pyyaml",
-    "requests",
-    "tabulate",
-    "tenacity",
-    "termcolor",
-    "tqdm",
-    "xmltodict",
-    "typing-extensions"
-]
 
-[tool.flit.sdist]
-include = ["resources/"]
+[tool.poetry.dependencies]
+python = "^3.6.2"
 
-[project.urls]
-Source = "https://github.com/yandex/ch-tools"
+boto3 = "~1.23.10"
+click = "~8.0.4"
+cloup = "*"
+deepdiff = "*"
+dnspython = "*"
+file_read_backwards = "*"
+humanfriendly = "*"
+jinja2 = "*"
+kazoo = "*"
+lxml = "*"
+psutil = "*"
+pygments = "*"
+pyopenssl = "*"
+python-dateutil = "*"
+pyyaml = "<5.4"
+requests = "*"
+tabulate = "*"
+tenacity = "*"
+termcolor = "*"
+tqdm = "*"
+typing-extensions = "~4.1.1"
+xmltodict = "*"
 
-[project.optional-dependencies]
-test = [
-    "flit",
-    "black",
-    "isort",
-    "flake8",
-    "flake8-string-format",
-    "flake8-print",
-    "pylint",
-    "mypy",
-    "types-pyyaml",
-    "types-requests",
-    "types-python-dateutil",
-    "types-tabulate",
-    "types-pyOpenSSL",
-    "types-setuptools",
-    "behave",
-    "docker",
-    "docker-compose",
-    "pyhamcrest",
-    "pytest",
-    "flask"
-]
+[tool.poetry.dev-dependencies]
+behave = "*"
+black = { version = "^22.0.0", python = ">=3.10" }
+docker = "*"
+docker-compose = "*"
+flake8 = { version = "^6.1.0", python = ">=3.10" }
+isort = { version = "^5.0.0", extras = ["pyproject"] }
+mypy = { version = "^1.4.1", python = ">=3.10" }
+pyhamcrest = "*"
+pylint = { version = "^2.17.5", python = ">=3.10" }
+pytest = "*"
+types-pyOpenSSL = "*"
+types-python-dateutil = "*"
+types-pyyaml = "*"
+types-requests = "*"
+types-setuptools = "*"
+types-tabulate = "*"
```

Comment:
`types-*` packages don't make much sense without `mypy`. So it's worth to add `python = ">=3.10"` for `types-*` packages too. This should allow to use up-to-date versions of `pyyaml`, `boto` and `click`. At least in ch-backup it helps - https://github.com/yandex/ch-backup/pull/26/commits/8a02cad113160ddfb2e4f4f451f44e3c87071121#diff-2b4945591edfeaa4cf4d3f155e66d4b43d1bda7a55d881d5cf3107f1b05abbbc / https://github.com/yandex/ch-backup/pull/26/

Reply:
Agree. Did it.